bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/DataIterator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.jvm;
+
+import org.ballerinalang.jvm.types.BStructureType;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ *  This interface represents an data iterator operations.
+ *  Each data source need to implement their own implementation by implementing this interface.
+ *  Known implementations: {@code SQLDataIterator}
+ *
+ *  @since 0.995.0
+ */
+public interface DataIterator {
```

Comment:
With the changes of #15235, `DataIterator` needs to be extended from `IteratorValue`.

Reply:
Will fix it once #15235 is merged